tests_integ/mcp/test_mcp_client.py

```diff
@@ -297,3 +302,27 @@ def slow_transport():
     with client:
         tools = client.list_tools_sync()
         assert len(tools) >= 0  # Should work now
+
+
+@pytest.mark.skipif(
+    condition=os.environ.get("GITHUB_ACTIONS") == "true",
+    reason="streamable transport is failing in GitHub actions, debugging if linux compatibility issue",
+)
+@pytest.mark.asyncio
+async def test_streamable_http_mcp_client_times_out_before_tool():
+    """Test an mcp server that timesout before the tool is able to respond."""
+    server_thread = threading.Thread(
+        target=start_comprehensive_mcp_server, kwargs={"transport": "streamable-http", "port": 8001}, daemon=True
+    )
+    server_thread.start()
+    time.sleep(2)  # wait for server to startup completely
```

Comment:
I presume there's no way to mock time or do this in a way that doesn't cause a constant delay when running tests?